src/gcp/cloudsql/connect.ts

```diff
@@ -105,82 +106,102 @@ export async function execute(
   await cloudSqlAdminClient.createUser(
     projectId,
     instanceId,
     "BUILT_IN",
-    setupUser,
+    superuser,
     temporaryPassword,
   );
 
-  // 2. Create an IAM user for the current identity
-  const { user, mode } = toDatabaseUser(account);
-  await cloudSqlAdminClient.createUser(projectId, instanceId, mode, user);
-
-  // 3. Connect to the DB as the temporary user and run the necessary grants
-  // TODO: I think we're missing something here, sometimes backend can't see the tables.
-  const grants = [
-    `do
-      $$
-      begin
-        if not exists (select FROM pg_catalog.pg_roles
-          WHERE  rolname = '${firebaseowner(databaseId)}') then
-          CREATE ROLE "${firebaseowner(databaseId)}" WITH ADMIN "${setupUser}";
-        end if;
-      end
-      $$
-      ;`,
-    `GRANT ALL PRIVILEGES ON DATABASE "${databaseId}" TO "${firebaseowner(databaseId)}"`,
-    `GRANT cloudsqlsuperuser TO "${firebaseowner(databaseId)}"`,
-    `GRANT "${firebaseowner(databaseId)}" TO "${setupUser}"`,
-    `GRANT "${firebaseowner(databaseId)}" TO "${user}"`,
-    `ALTER SCHEMA public OWNER TO "${firebaseowner(databaseId)}"`,
-    `GRANT USAGE ON SCHEMA "public" TO PUBLIC`,
-    `GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA "public" TO PUBLIC`,
-    `GRANT ALL PRIVILEGES ON ALL SEQUENCES IN SCHEMA "public" TO PUBLIC`,
-  ];
-  await execute(grants, {
+  return await execute([`SET ROLE = cloudsqlsuperuser`, ...cmds], {
     projectId,
     instanceId,
     databaseId,
-    username: setupUser,
+    username: superuser,
     password: temporaryPassword,
-    silent: true,
+    silent: silent,
   });
-  return user;
 }
 
-export function firebaseowner(databaseId: string) {
-  return `firebaseowner_${databaseId}_public`;
+export async function getIAMUser(options: Options): Promise<{ user: string; mode: UserType }> {
+  const account = await requireAuth(options);
+  if (!account) {
+    throw new FirebaseError(
+      "No account to set up! Run `firebase login` or set Application Default Credentials",
+    );
+  }
+
+  return toDatabaseUser(account);
+}
+
+// setupIAMUser sets up the current user identity to connect to CloudSQL.
+// Steps:
+// 2. Create an IAM user for the current identity
+// 3. Connect to the DB as the temporary user and run the necessary grants
+// 4. Deletes the temporary user
+export async function setupIAMUser(
+  instanceId: string,
+  databaseId: string,
+  options: Options,
+): Promise<string> {
+  // TODO: Is there a good way to short circuit this by checking if the IAM user exists and has the appropriate role first?
+  const projectId = needProjectId(options);
+
+  // 0. Get the current identity
+  const { user, mode } = await getIAMUser(options);
+
+  // 1. Create an IAM user for the current identity.
+  await cloudSqlAdminClient.createUser(projectId, instanceId, mode, user);
+
+  // 2. Setup FDC required SQL roles and permissions.
+  await setupSQLPermissions(instanceId, databaseId, options, true);
```

Comment:
`silent = true`

In general, literal argument should be named. Hard to guess what it's for.

Reply:
It's not possible to name arguments in typescript/javascript afaik. Do you mean creating a variable for it `const silent = true`?

Reply:
Added a gsdoc comment